stdlib/list.mli

```diff
@@ -102,6 +102,39 @@ val flatten : 'a list list -> 'a list
 (** An alias for [concat]. *)
 
 
+(** {1 Comparison} *)
+
+val equal : ('a -> 'a -> bool) -> 'a list -> 'a list -> bool
+(** [equal eq [a1; ...; an] [b1; ..; bm]] holds when
+    the two input lists have the same length, and for each
+    pair of elements [ai], [bi] at the same position we have
```

Comment:
Nitpicking: `at the same position` superfluous given the notation.

Reply:
I think that having the mention may make things more accessible for people that are less familiar with mathematical notation.

Reply:
On the other hand it makes it harder to understand/more confusing for those who know. Drawing from other doc strings below you could say:
```
[equal eq [a1; ...; an] [b1; ...; bn]] is [eq a1 b1 && eq a2 b2 && ... && eq an bn] or 
[false] if the lists are not of the same size. 
```

Reply:
Great suggestion, thanks.